src/aarch64/ffi.c

```diff
@@ -728,9 +734,18 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
   else if (flags & AARCH64_RET_NEED_COPY)
     rsize = 16;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+  stack_padding = __builtin_cheri_round_representable_length(stack_bytes) -
+    stack_bytes;
+#endif
+
   /* Allocate consectutive stack for everything we'll need.
      The frame uses 40/80 bytes for: lr, fp, rvalue, flags, sp */
-  context = alloca (sizeof(struct call_context) + stack_bytes + sizeof(struct call_frame) + rsize);
+  context = alloca (sizeof(struct call_context) +
+#ifdef __CHERI_PURE_CAPABILITY__
+		    stack_padding +
```

Comment:
Just because the length is representable doesn't mean the bounds will be at an offset of `sizeof(struct call_context)` within the allocation (only guaranteed that the whole allocation is).

Reply:
(In an extreme, consider sizeof(struct call_context) is 1, stack_bytes is 2^32, then there's no padding but your requested base for the subregion will have an LSB of 1, which isn't representable for a length of 2^32)

Reply:
I think that further adding `~__builtin_cheri_representable_alignment_mask(stack_bytes)` to `stack_padding` would be a simple fix to this, albeit wasting some space.

Reply:
Oh, but then this would cause the `struct call_frame` to be completely misaligned. We should probably add `~__builtin_cheri_representable_alignment_mask(stack_bytes) + 1` instead so that everything after `stack_bytes` is at least 16-byte aligned.

Reply:
Well, this is still not quite right. If `stack_bytes` is sufficiently small (say 16) then the padding would be 1 and `struct call_frame` would still be misaligned. We should probably write `__align_up(~__builtin_cheri_representable_alignment_mask(stack_bytes), _Alignof(struct call_frame))` instead.